azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/implementation/SiteImpl.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    @Override
+    public DateTime lastModifiedTimeUtc() {
+        return inner().lastModifiedTimeUtc();
+    }
+
+    @Override
+    public SiteConfigInner siteConfig() {
+        return inner().siteConfig();
+    }
+
+    @Override
+    public List<String> trafficManagerHostNames() {
+        return inner().trafficManagerHostNames();
+    }
+
+    @Override
+    public boolean premiumAppDeployed() {
+        return inner().premiumAppDeployed();
+    }
+
+    @Override
+    public boolean scmSiteAlsoStopped() {
+        return inner().scmSiteAlsoStopped();
+    }
+
+    @Override
+    public String targetSwapSlot() {
+        return inner().targetSwapSlot();
+    }
+
+    @Override
+    public HostingEnvironmentProfile hostingEnvironmentProfile() {
+        return inner().hostingEnvironmentProfile();
+    }
+
+    @Override
+    public String microService() {
+        return inner().microService();
+    }
+
+    @Override
+    public String gatewaySiteName() {
+        return inner().gatewaySiteName();
+    }
+
+    @Override
+    public boolean clientAffinityEnabled() {
+        return inner().clientAffinityEnabled();
+    }
+
+    @Override
+    public boolean clientCertEnabled() {
+        return inner().clientCertEnabled();
+    }
+
+    @Override
+    public boolean hostNamesDisabled() {
+        return false;
+    }
+
+    @Override
+    public String outboundIpAddresses() {
+        return inner().outboundIpAddresses();
+    }
+
+    @Override
+    public int containerSize() {
+        return inner().containerSize();
+    }
+
+    @Override
+    public int maxNumberOfWorkers() {
+        return inner().maxNumberOfWorkers();
+    }
+
+    @Override
+    public CloningInfo cloningInfo() {
+        return inner().cloningInfo();
+    }
+
+    @Override
+    public String resourceGroup() {
+        return inner().resourceGroup();
+    }
+
+    @Override
+    public boolean isDefaultContainer() {
+        return inner().isDefaultContainer();
+    }
+
+    @Override
+    public String defaultHostName() {
+        return inner().defaultHostName();
+    }
+
+    @Override
+    protected void createResource() throws Exception {
+        SiteInner site = client.createOrUpdateSite(resourceGroupName(), name(), inner()).getBody();
+        this.setInner(site);
+    }
+
+    @Override
+    protected ServiceCall createResourceAsync(ServiceCallback<Void> callback) {
+        return client.createOrUpdateSiteAsync(resourceGroupName(), name(), inner(), Utils.fromVoidCallback(this, callback));
+    }
+
+    @Override
+    public Site refresh() throws Exception {
+        this.setInner(client.getSite(resourceGroupName(), name()).getBody());
+        return this;
+    }
+
+    @Override
+    public Site.DefinitionStages.WithCreate withNewAppServicePlan() {
+        return null;
+    }
+
+    @Override
+    public Site.DefinitionStages.WithCreate withNewAppServicePlan(String name, AppServicePricingTier pricingTier) {
```

Comment:
generally, whenever we can, return the impl itself, not the specific interface the impl is implementing, that way we separate the user-facinmg logic from the impl and enable ourselves to use the same impl'd method to implement multiple flulent interfaces (Definition & Update) in one go. So these withers should consistently return SiteImpl.